TypeCobol.LanguageServer/Completion Factory/CompletionFactoryHelpers.cs

```diff
@@ -45,8 +45,7 @@ public static Node GetMatchingNode(FileCompiler fileCompiler, CodeElement codeEl
             {
                 return node;
             }
-
-            return null;
+            return codeElementToNode.FirstOrDefault(t => t.Key.Equals(codeElement)).Value;
```

Comment:
What is that ?? At this point `codeElementToNode` maybe null and if it is not, it does not contain the entry for `codeElement` key, so the `FirstOrDefault` will return a default `KeyValuePair` leading to a `null` value return.
Please revert.

Reply:
I agree codeElementToNode can be null, thus I am going to replaced this line by:
codeElementToNode?.FirstOrDefault(t => t.Key.Equals(codeElement)).Value;
Without that line it does not work.

Reply:
Sorry I don't know why I came to this conclusion, I retested again, you are right it must be reverted.

Reply:
I've run tests (everything except ThirdParty) in EI_Debug configuration on your branch replacing this line with the original `return null;` and it went fine. Can you point out the case where it won't work ?
The dictionary uses  the default comparer so it relies on `GetHashCode` and `Equals` methods of code elements. If this line does not return `null`, it means that `GetHashCode` and `Equals` are not synced for some CE.

Reply:
Yes it will be reverted

Reply:
done in 6a515c8